src/zabapgit_object_form.prog.abap

```diff
@@ -0,0 +1,330 @@
+
+    TYPES: tyt_header TYPE STANDARD TABLE OF thead WITH DEFAULT KEY.
+    TYPES: tys_header TYPE LINE OF tyt_header.
+    TYPES: tyt_lines  TYPE tline_tab.
+
+    METHODS _get_last_changes
+      IMPORTING
+        iv_form_name           TYPE ty_item-obj_name
+      RETURNING
+        VALUE(es_last_changed) TYPE thead.
+
+    METHODS _build_extr_from_header
+      IMPORTING
+        ls_header       TYPE tys_header
+      RETURNING
+        VALUE(r_result) TYPE string.
+
+ENDCLASS.
+
+*----------------------------------------------------------------------*
+*       CLASS lcl_object_form IMPLEMENTATION
+*----------------------------------------------------------------------*
+CLASS lcl_object_form IMPLEMENTATION.
+
+  METHOD lif_object~has_changed_since.
+
+    DATA: ls_last_changed TYPE thead.
+    DATA: lv_last_changed_ts TYPE timestamp.
+
+    ls_last_changed = _get_last_changes( ms_item-obj_name ).
+
+    CONVERT DATE ls_last_changed-tdldate TIME ls_last_changed-tdltime
+            INTO TIME STAMP lv_last_changed_ts TIME ZONE sy-zonlo.
+
+    rv_changed = boolc( sy-subrc <> 0 OR lv_last_changed_ts > iv_timestamp ).
+
+  ENDMETHOD.
+
+  METHOD lif_object~changed_by.
+
+    DATA: ls_last_changed TYPE thead.
+
+    ls_last_changed = _get_last_changes( ms_item-obj_name ).
+
+    IF ls_last_changed-tdluser IS NOT INITIAL.
+      rv_user = ls_last_changed-tdluser.
+    ELSE.
+      rv_user = c_user_unknown.
+    ENDIF.
+
+  ENDMETHOD.
+
+  METHOD lif_object~get_metadata.
+
+    rs_metadata = get_metadata( ).
+    rs_metadata-delete_tadir = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~exists.
+
+    DATA: lv_form_name TYPE thead-tdform.
+
+    lv_form_name = ms_item-obj_name.
+
+    CALL FUNCTION 'READ_FORM'
+      EXPORTING
+        form  = lv_form_name
+      IMPORTING
+        found = rv_bool.
+
+  ENDMETHOD.
+
+  METHOD lif_object~jump.
+
+    DATA: lt_bdcdata TYPE TABLE OF bdcdata.
+
+    FIELD-SYMBOLS: <ls_bdcdata> LIKE LINE OF lt_bdcdata.
+
+    APPEND INITIAL LINE TO lt_bdcdata ASSIGNING <ls_bdcdata>.
+    <ls_bdcdata>-program  = 'SAPMSSCF' ##NO_TEXT.
+    <ls_bdcdata>-dynpro   = '1102' ##NO_TEXT.
+    <ls_bdcdata>-dynbegin = abap_true.
+
+    APPEND INITIAL LINE TO lt_bdcdata ASSIGNING <ls_bdcdata>.
+    <ls_bdcdata>-fnam = 'BDC_OKCODE' ##NO_TEXT.
+    <ls_bdcdata>-fval = '=SHOW' ##NO_TEXT.
+
+    APPEND INITIAL LINE TO lt_bdcdata ASSIGNING <ls_bdcdata>.
+    <ls_bdcdata>-fnam = 'RSSCF-TDFORM' ##NO_TEXT.
+    <ls_bdcdata>-fval = ms_item-obj_name.
+
+    CALL FUNCTION 'ABAP4_CALL_TRANSACTION'
+      STARTING NEW TASK 'GIT'
+      EXPORTING
+        tcode     = 'SE71'
+        mode_val  = 'E'
+      TABLES
+        using_tab = lt_bdcdata
+      EXCEPTIONS
+        OTHERS    = 1
+        ##fm_subrc_ok.                                                   "#EC CI_SUBRC
+
+  ENDMETHOD.
+
+  METHOD lif_object~delete.
+
+    DATA: lv_name TYPE thead-tdname.
+
+    lv_name = ms_item-obj_name.
+
+    CALL FUNCTION 'DELETE_TEXT'
```

Comment:
with the FORM FMs the data is structured, so if someone needs to use the data outside of an ABAP system it will be more convenient if the data is already parsed